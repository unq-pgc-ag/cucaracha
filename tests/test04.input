
fun conj(a : Bool, b : Bool) : Bool {
  if x {
    if y {
      z := True
    } else {
      z := False
    }
  } else {
    z := False
  }
  return z
}

fun main() {
  v := [0,1]
  i := 0
  while i < 2 {
    j := 0
    while j < 2 {
      if conj(v[i] == 0, v[j] == 0) {
        putNum(1) putChar(10)
      } else {
        putNum(0) putChar(10)
      }
      j := j + 1
    }
    i := i + 1
  }
}

