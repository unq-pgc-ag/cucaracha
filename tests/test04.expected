(Program
  (Function
    conj
    Bool
    (Parameter
      a
      Bool
    )
    (Parameter
      b
      Bool
    )
    (Block
      (StmtIfElse
        (ExprVar
          x
        )
        (Block
          (StmtIfElse
            (ExprVar
              y
            )
            (Block
              (StmtAssign
                z
                (ExprConstBool
                  True
                )
              )
            )
            (Block
              (StmtAssign
                z
                (ExprConstBool
                  False
                )
              )
            )
          )
        )
        (Block
          (StmtAssign
            z
            (ExprConstBool
              False
            )
          )
        )
      )
      (StmtReturn
        (ExprVar
          z
        )
      )
    )
  )
  (Function
    main
    Unit
    (Block
      (StmtAssign
        v
        (ExprVecMake
          (ExprConstNum
            0
          )
          (ExprConstNum
            1
          )
        )
      )
      (StmtAssign
        i
        (ExprConstNum
          0
        )
      )
      (StmtWhile
        (ExprLt
          (ExprVar
            i
          )
          (ExprConstNum
            2
          )
        )
        (Block
          (StmtAssign
            j
            (ExprConstNum
              0
            )
          )
          (StmtWhile
            (ExprLt
              (ExprVar
                j
              )
              (ExprConstNum
                2
              )
            )
            (Block
              (StmtIfElse
                (ExprCall
                  conj
                  (ExprEq
                    (ExprVecDeref
                      v
                      (ExprVar
                        i
                      )
                    )
                    (ExprConstNum
                      0
                    )
                  )
                  (ExprEq
                    (ExprVecDeref
                      v
                      (ExprVar
                        j
                      )
                    )
                    (ExprConstNum
                      0
                    )
                  )
                )
                (Block
                  (StmtCall
                    putNum
                    (ExprConstNum
                      1
                    )
                  )
                  (StmtCall
                    putChar
                    (ExprConstNum
                      10
                    )
                  )
                )
                (Block
                  (StmtCall
                    putNum
                    (ExprConstNum
                      0
                    )
                  )
                  (StmtCall
                    putChar
                    (ExprConstNum
                      10
                    )
                  )
                )
              )
              (StmtAssign
                j
                (ExprAdd
                  (ExprVar
                    j
                  )
                  (ExprConstNum
                    1
                  )
                )
              )
            )
          )
          (StmtAssign
            i
            (ExprAdd
              (ExprVar
                i
              )
              (ExprConstNum
                1
              )
            )
          )
        )
      )
    )
  )
)
